### DISCUSSION TITLE
Subcommand for validation

### DISCUSSION THREAD

**Discussion by jelovirt** [dita-ot/dita-ot#4398] [Ideas]
I opened an issue for adding a subcommand to `dita` command to just do validation. Please check https://github.com/dita-ot/dita-ot/issues/4397 and add your comments and ideas.

**Comment by stefan-jung** [comment]
It would be nice, if it would be possible to pass a set of Schematron files to the validation process.

And it would be cool to have a "validation" XSLT mode in which we could trigger templates which match on certain conditions to spawn warning and info messages. This would be a non-Schematron pure XSLT validation mechanism.

**Reply by jelovirt** [reply]
> It would be nice, if it would be possible to pass a set of Schematron files to the validation process.

Because `validate` would be a magic name for transtype, the idea is that third-party plug-ins could run e.g. Schematron validation. This way the validation subcommand would not implement this directly.

> And it would be cool to have a "validation" XSLT mode in which we could trigger templates which match on certain conditions to spawn warning and info messages. This would be a non-Schematron pure XSLT validation mechanism.

Again, this would be left up to third-party plugins that could extend the validation subcommand.

**Comment by stefan-jung** [comment]
Another idea, which is related to https://github.com/dita-ot/docs/issues/539: It would be nice, if the DITA-OT would "validate"/check if none of the processed (temporary) files exceeds the file path limitations of the file system. To simplify this, it would be possible to take 260 as a threshold, because it would probably not possible to add filesystem specific validation. If a filepath is too long, the DITA-OT could trigger a warning.
